graphql.go

```diff
@@ -32,6 +32,7 @@ func ParseSchema(schemaString string, resolver interface{}, opts ...SchemaOpt) (
 		tracer:         noop.Tracer{},
 		logger:         &log.DefaultLogger{},
 		panicHandler:   &errors.DefaultPanicHandler{},
+		directives:     map[string]directives.ResolverVisitor{},
```

Comment:
I think that we might have to differentiate between different directives. Maybe directives should be a config object:
```go
type Schema struct {
    ...
    directives: &Driectives{...},
    ...
}

// Directives contains maps of directive visitors for different directive locations.
// https://spec.graphql.org/draft/#DirectiveLocation
type Directives struct {
    FieldDefinition map[string]directives.ResolverVisitor{}  // ON FIELD_DEFINITION
    Field           map[string]directives.FieldVisitor{}     // ON FIELD, FRAGMENT_SPREAD, INLINE_FRAGMENT
    Argument        map[string]directives.ArgumentResolver{} // ON ARGUMENT_DEFINITION
    Variable        map[string]directives.VariableVisitor{}  // ON VARIABLE_DEFINITION
    Schema          map[string]directives.SchemaDirective{}
    Scalar          map[string]directives.ScalarDirective{}
    // more directive locations here
}
```
With the above, if we add more fields (i.e. locations) to `schema.Directives` we wouldn't break the API backwards compatibility.
Now that there is at least one directive packer created it would pave the way for more types of directives to be implemented. For example, input validation like `@isEmail`, `@matchRegex(regex: String!)`, `@inRange(min: Int, max: Int)` etc. It makes sense to reuse validation rules. I mean that different directives would do different things based on their location. Obviously, a field definition directive wraps the field resolver, however, a schema or argument directive would do something completely different (their interfaces would differ).